### PR TITLE
Fix many-to-many wrong table for order column

### PR DESCRIPTION
**Pull request by dreab8**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/ordering/ast/ColumnReference.java [hunk 1]
```diff
@@ -110,7 +109,11 @@ TableReference getTableReference(TableGroup tableGroup) {
 		ModelPartContainer modelPart = tableGroup.getModelPart();
 		if ( modelPart instanceof PluralAttributeMapping ) {
 			final PluralAttributeMapping pluralAttribute = (PluralAttributeMapping) modelPart;
-			final MappingType elementMappingType = (pluralAttribute).getElementDescriptor().getPartMappingType();
+			if ( !pluralAttribute.getCollectionDescriptor().hasManyToManyOrdering() ) {
+				return tableGroup.getPrimaryTableReference();
+			}
+
+			final MappingType elementMappingType = ( pluralAttribute ).getElementDescriptor().getPartMappingType();
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/metamodel/mapping/ordering/ast/ColumnReference.java:116]:
```suggestion
			final MappingType elementMappingType = pluralAttribute.getElementDescriptor().getPartMappingType();
```

**Review by beikov — APPROVED** [review]
LGTM!

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/ordering/ast/ColumnReference.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/metamodel/mapping/ordering/ast/ColumnReference.java]: Thanks @beikov, fixed it

File: hibernate-core/src/test/java/org/hibernate/orm/test/idbag/IdBagTest.java [hunk 2]
```diff
@@ -76,33 +73,33 @@ public void testUpdateIdBag() throws HibernateException {
...
+
+					List l = session.createQuery( "from User u join u.groups g" ).list();
+					assertEquals( 1, l.size() );
+
+					session.clear();
+
+					gavin = (User) session.createQuery( "from User u join fetch u.groups" ).uniqueResult();
+					assertTrue( Hibernate.isInitialized( gavin.getGroups() ) );
+					assertEquals( gavin.getGroups().size(), 2 );
+					assertEquals( ( (Group) gavin.getGroups().get( 0 ) ).getName(), "admins" );
```
Comment (NathanQingyangXu) [inline hibernate-core/src/test/java/org/hibernate/orm/test/idbag/IdBagTest.java:96]: technically, you messed up the parameter order, though the difference can only be noticed when the testing case is broken.

Comment (dreab8) [reply hibernate-core/src/test/java/org/hibernate/orm/test/idbag/IdBagTest.java:96]: I haven't noticed that the original was using the wrong order, I'm going to fix it.
Thanks

**Comment by NathanQingyangXu** [comment]
Arguably hamcrest assertThat is more readable, consistent and free of such
issue. But it seems not mandatory.

**Comment by beikov** [comment]
Don't worry about the failure for MariaDB. wip/6.0 fails also due to a recent test move by Steve. I fixed that test on my other PR already though. IMO we can merge this.

**Comment by dreab8** [comment]
Applied upstream,

Thanks @beikov and @NathanQingyangXu for the review!

**Pull request closed without merging**